### DISCUSSION TITLE
Does signing with avbroot works with device integrity?

### DISCUSSION THREAD

**Discussion by HpNoTiQ56** [chenxiaolong/avbroot#380] [Q&A]
Hi,
I wanted to ask you how to patched my ota files in order to maintain device integrity? If you use AOSP certificate on stock image with trickystore, you can have device integrity but I'm looking to get it using my own custom key (not strong). Is it possible?

**Comment by chenxiaolong** [comment]
Sorry, I'm not at all familiar with Play Integrity. I don't use anything that requires passing it.

Google knows which key the OS is signed with via hardware attestation. I don't know if that can be spoofed.

**Reply by HpNoTiQ56** [reply]
Seems like can't pass device but RCS works.
